src/Couchbase.Lite.Shared/API/Document/MutableDocument.cs

```diff
@@ -347,6 +361,24 @@ public IMutableDictionary SetDictionary(string key, DictionaryObject value)
             return this;
         }
 
+        /// <inheritdoc />
+        public IMutableDictionary SetJSON([NotNull] string json)
+        {
+            JObject jobj = null;
+            try {
+                jobj = JObject.Parse(json);
+            } catch {
+                throw new CouchbaseLiteException(C4ErrorCode.InvalidParameter, CouchbaseLiteErrorMessage.InvalidJSON);
+            }
+
+            if (jobj.GetType() == typeof(JArray)) {
```

Comment:
Should it check if it's not dictionary instead? The other types such as string and numbers are also valid JSON.

Reply:
Line 378 below will deserialize json string into Dictionary<string, object> object, which will make the validation there.
JObject.Parse(json); checks if string passed in is json format, and jobj  is the json object if json successfully parsed. Then we will need to make sure jobj is not an array type before we deserialize the json string into dictionary object.

Reply:
What will happen if it is just a string or number like Pasin says though?

Reply:
It throws invalid argument. I just added couple new tests TestMutableDictSetJsonWithInvalidParam & TestMutableDocumentSetJsonWithInvalidParam to prove.

Reply:
👍